### PR TITLE
Ci across os

### PR DESCRIPTION
**Pull request by yinweisu**
*Description of changes:*
Have github action ci to cover cpu tests. This enables us to easily test on multiple platforms and python versions

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1445-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1445/4/index.html

**Comment by szha** [comment]
Job PR-1445-5 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1445/5/index.html

**Comment by yinweisu** [comment]
There's currently one issue with this PR: Failing a github action won't stop jenkins CI from executing. Haven't found a solution to it yet.

File: .github/workflows/cpu_unittest.yml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        run: |
+          conda env create -n unittest -f .github/workflows_env/unittest_env.yml
+          conda activate unittest
+          conda list
+          python3 -m pip install --upgrade -e common/[tests]
+          python3 -m pip install --upgrade -e core/[all]
+          python3 -m pip install --upgrade -e features/
+          python3 -m pip install --upgrade -e tabular/[all]
+          python3 -m pip install --upgrade -e text/
+          python3 -m pip install --upgrade -e vision/
```
Comment (gradientsky) [inline .github/workflows/cpu_unittest.yml:123]: not sure if we need this for `tabular`

Comment (yinweisu) [reply .github/workflows/cpu_unittest.yml:123]: We are offloading cpu tests to github action, that's why I put tabular here

Comment (Innixma) [reply .github/workflows/cpu_unittest.yml:123]: Alex is referring to the installation of text and vision modules

**Review by Innixma — APPROVED** [review]
Looks awesome, great stuff!

File: .github/workflows/cpu_unittest.yml (hunk 1 above)
Comment (yinweisu) [reply .github/workflows/cpu_unittest.yml:123]: Oh, the tabular CI will fail while installing autogluon without text and vision

Comment (Innixma) [reply .github/workflows/cpu_unittest.yml:123]: Why? Isn't this CPU only?

Comment (yinweisu) [reply .github/workflows/cpu_unittest.yml]: NVM, that's actually a mistake. Updated it

**Comment by szha** [comment]
Job PR-1445-6 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1445/6/index.html

**Comment by yinweisu** [comment]
@ci_cross_platforms

**Comment by yinweisu** [comment]
Close because this change can only be tested on a separate branch instead of a fork. Will create a new PR

**Pull request closed without merging**